### PR TITLE
PLT-6828 - Removing urls for links

### PR DESCRIPTION
**Pull request by asaadmahmood**
#### Summary
PLT-6828 - Removing urls for links

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6828

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has server changes (please link)
- [ ] Has redux changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

**Comment by jasonblais** [comment]
@asaadmahmood Can you help resolve the merge conflicts and rebase against the latest `master` branch? I'll spin up a test server after that.

**Comment by asaadmahmood** [comment]
1. Talked to @hmhealey, and we wouldn't want to hide the url bar in the sidebar, as those are actual links to other pages. Or if you mean that we should still have them as links and just hide the url bar, then instead of doing this, and if we don't even want the url bar in general for normal links as well, then we can just try to look into doing that for all links rather than this change.
2. Will fix that.
3. Cool, will fix.
4. Cool, will fix.
5. Cool, will fix.

**Comment by asaadmahmood** [comment]
@jasonblais You can test it out again, fixed items 2-5.

**Comment by jasonblais** [comment]
@asaadmahmood 

How much mana is involved in the second approach where we hide the URL bar?

**Comment by jasonblais** [comment]
For 1), discussed with Harrison & Asaad: We'll add a special case for desktop app where channels and teams in the sidebar are treated as buttons.

I will ask Linda to help test this next, but will wait until next week as we're preparing for v4.3 release testing.

**Comment by lindy65** [comment]
Tested on Win7, Chrome.

1. Hovering over the `flag icon` in the center channel, RHS or to the right of "search input box" shows a URL
image

2. Hovering over the `timestamp` in the center channel or RHS show a URL
image

3. `Team Invite Link` from Main Menu > Send Email Invite > Team Invite Link is no longer linked.
image

4. Hovering over `General`, `Import` or `Regenerate` in Main Menu > Team Settings > (Invite Code) shows a URL 
image

5. Hovering over `Member` in Main Menu > Manage Members shows a URL
image

**Comment by jasonblais** [comment]
Thanks @lindy65, this is great!

I reviewed my notes from above and only have this one to add: 

6) "Team Invite Link" in "Send Email Invite" dialog is no longer clickable.
image

**Comment by asaadmahmood** [comment]
@jasonblais @lindy65 
1. Done.
2. That's valid, as it takes you to a different link.
3. Done.
4. Done.
5. Done.
6. That looks like the same as 3.

File: components/channel_header/channel_header.jsx [hunk 1]
```diff
@@ -111,9 +111,7 @@ export default class ChannelHeader extends React.Component {
         }
     }
 
-    toggleFavorite = (e) => {
-        e.preventDefault();
```
Comment (hmhealey) [inline components/channel_header/channel_header.jsx]: This is going to be really nice to get rid of all these preventDefault calls all over the place

**Comment by asaadmahmood** [comment]
@hmhealey Thanks for the pointer. Redid the changes so that now we use the `<button>` tag instead of divs or spans.

**Pull request merged**